File: packages/gbnf-experiment/data/20260909T015142Z_f8ca89ff/ported_implementation/grammar_graph/get_input_as_code_points.py
from __future__ import annotations

from typing import TYPE_CHECKING, List

if TYPE_CHECKING:  # pragma: no cover
    from .types import ValidInput

__all__ = ["get_input_as_code_points", "getInputAsCodePoints"]


def get_input_as_code_points(src: "ValidInput") -> List[int]:
    if not isinstance(src, str):
        return list(src) if isinstance(src, (list, tuple)) else [src]

    return [ord(char) for char in src]


getInputAsCodePoints = get_input_as_code_points

File: proposed-projects/round-trip-experiment/reverse/20260909T140517Z_670261ce/20260910T081441Z_4899d5cf/ported_implementation/src/grammar-graph/get-input-as-code-points.ts
import type { ValidInput } from './types.js';

export const getInputAsCodePoints = (src: ValidInput): number[] => {
  if (typeof src !== 'string') {
    return Array.isArray(src) ? [...src] : [src];
  }

  return Array.from(src, (char) => char.codePointAt(0) as number);
};
